app/controllers/dashboards_controller.rb

```diff
@@ -32,15 +32,22 @@ def show
 
   def pro
     authorize current_user, :pro_user?
-    @this_week_reactions_count = Reaction.where(reactable_id: current_user.articles.pluck(:id), reactable_type: "Article").where("created_at > ?", 1.week.ago).size
-    @last_week_reactions_count = Reaction.where(reactable_id: current_user.articles.pluck(:id), reactable_type: "Article").where("created_at > ? AND created_at < ?", 2.weeks.ago, 1.week.ago).size
-    @this_month_reactions_count = Reaction.where(reactable_id: current_user.articles.pluck(:id), reactable_type: "Article").where("created_at > ?", 1.month.ago).size
-    @last_month_reactions_count = Reaction.where(reactable_id: current_user.articles.pluck(:id), reactable_type: "Article").where("created_at > ? AND created_at < ?", 2.months.ago, 1.months.ago).size
+    @current_user_article_ids = current_user.articles.pluck(:id)
```

Comment:
I suggest using a facade pattern here instead of using lots of instance variables.
Like creating a class `Dashboards::Pro` (or another name) with the corresponding methods.
[Example](https://thoughtbot.com/blog/sandi-metz-rules-for-developers#only-instantiate-one-object-in-the-controller)

Reply:
+ that would allow to add a separate test for the data easier.

Reply:
This makes sense. @Zhao-Andy refactoring as such sooner rather than later is probably good.